Engine/DataFeeds/TimeSlice.cs

```diff
@@ -380,6 +380,20 @@ private static bool HandleOptionData(DateTime algorithmTime, BaseData baseData,
 
                 case MarketDataType.QuoteBar:
                     var quote = (QuoteBar)baseData;
+
+                    // if fill-forward enabled, copy previous bid/ask when current values are null
+                    if (security.IsFillDataForward)
+                    {
+                        if (quote.Bid == null)
+                        {
+                            quote.Bid = new Bar(contract.BidPrice, contract.BidPrice, contract.BidPrice, contract.BidPrice);
```

Comment:
This fill forward should probably be done in the subscription enumerator stack somewhere; so we can pull in the full OHLC bar of the bid/ask objects and not just the current price. 

It would also stop the FF from polluting the time series create pathway.